pkg/operator/hierarchy/selector.go

```diff
@@ -79,7 +78,6 @@ var _ Selector = (*KeySelector)(nil)
 // ApplyToList implements Selector.
 func (ks *KeySelector) ApplyToList(lo *client.ListOptions) {
 	lo.Namespace = ks.Namespace
-	lo.FieldSelector = fields.OneTermEqualSelector("metadata.name", ks.Name)
```

Comment:
(Note that `Matches` below already performs filtering of the objects so we don't need to do anything other than remove the selector here)